src/Chapter4.hs

```diff
@@ -504,21 +504,22 @@ Implement the 'Applicative' instance for our 'List' type.
   type.
 -}
 {- ORMOLU_ENABLE -}
+-- copy pasta from chap3 instead of importing
+append :: List a -> List a -> List a
+append x y = merge y x
+  where
+    merge :: List a -> List a -> List a
+    merge acc Empty = acc
```

Comment:
You can add one more case to make your solution even more efficient:

```haskell
    merge Empty xs = xs
```

In that case, you won't need to walk through each element of the second list when the first one is empty and return the list instantly 🚤

Also, you can inline merge function directly into the append, if you will switch the arguments in the pattern matching 🙂

Reply:
OOOOOOOOOOOooooOOOOOOO, thanks